src/video-player.js

```diff
@@ -61,16 +61,38 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     super();
 
     this.nbCalls = 0;
-
     this.videoElement = getResolveVideoElement(elem);
 
-    this.options = extractOptions(this.videoElement, initOptions);
-
-    this._videojsOptions = this.options.videojsOptions;
-
     // Make sure to add 'video-js' class before creating videojs instance
     this.videoElement.classList.add('video-js');
 
+    // first phase - shouldn't be exposed to people until we have BE ready
+    const profileName = initOptions.__profileName;
+
+    if (profileName) {
+      this.initProfile(profileName, initOptions)
```

Comment:
Why passing `profileName, initOptions` when profileName is a member of initOptions?

Also you should probably pass this.options, and not just initOptions since initOptions doesn't go through extractOptions